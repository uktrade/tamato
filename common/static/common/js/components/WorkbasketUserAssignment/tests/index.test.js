import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";

import { WorkbasketUserAssignment } from "../index";

const mockUsers = [
  {
    pk: 1,
    name: "User One",
  },
  {
    pk: 2,
    name: "User Two",
  },
  {
    pk: 3,
    name: "User Three",
  },
];

describe("WorkbasketUserAssignment", () => {
  beforeAll(() => {
    window.CSRF_TOKEN = "abc123";
    window.assignUsersUrl = "/workbaskets/1/assign-users/";
    window.unassignUsersUrl = "/workbaskets/1/unassign-users/";
  });

  afterAll(() => {
    delete window.CSRF_TOKEN;
    delete window.assignUsersUrl;
    delete window.unassignUsersUrl;
  });

  afterEach(() => {
    document.body.innerHTML = "";
  });

  it("renders assign workers button", () => {
    const component = renderer.create(
      <WorkbasketUserAssignment
        action="Assign"
        assignment="workers"
        users={mockUsers}
        buttonId="assign-workers"
        formId="assign-workers-form"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders assign reviewers button", () => {
    const component = renderer.create(
      <WorkbasketUserAssignment
        action="Assign"
        assignment="reviewers"
        users={mockUsers}
        buttonId="assign-reviewers"
        formId="assign-reviewers-form"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders unassign workers button", () => {
    const component = renderer.create(
      <WorkbasketUserAssignment
        action="Unassign"
        assignment="workers"
        users={mockUsers}
        buttonId="assign-workers"
        formId="assign-workers-form"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders unassign reviewers button", () => {
    const component = renderer.create(
      <WorkbasketUserAssignment
        action="Unassign"
        assignment="reviewers"
        users={mockUsers}
        buttonId="unassign-reviewers"
        formId="unassign-reviewers-form"
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("creates form when button is clicked", () => {
    const assignmentRow = document.createElement("div");
    assignmentRow.classList.add("govuk-summary-list__row");

    render(
      <WorkbasketUserAssignment
        action="Assign"
        assignment="workers"
        users={mockUsers}
        buttonId="assign-workers"
        formId="assign-workers-form"
      />,
      {
        container: document.body.appendChild(assignmentRow),
      }
    );

    fireEvent.click(screen.getByTestId("assign-workers"));
    expect(screen.getByLabelText("Assign user")).toBeInTheDocument();
  });

  it("removes form when button is clicked twice", () => {
    const assignmentRow = document.createElement("div");
    assignmentRow.classList.add("govuk-summary-list__row");

    render(
      <WorkbasketUserAssignment
        action="Unassign"
        assignment="workers"
        users={mockUsers}
        buttonId="unassign-workers"
        formId="unassign-workers-form"
      />,
      {
        container: document.body.appendChild(assignmentRow),
      }
    );

    fireEvent.click(screen.getByTestId("unassign-workers"));
    expect(screen.getByLabelText("Unassign user")).toBeInTheDocument();

    fireEvent.click(screen.getByTestId("unassign-workers"));
    expect(screen.queryByLabelText("Unassign user")).not.toBeInTheDocument();
  });
});
