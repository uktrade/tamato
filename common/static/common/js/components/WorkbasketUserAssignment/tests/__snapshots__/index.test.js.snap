// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkbasketUserAssignment renders assign reviewers button 1`] = `
<button
  className="govuk-link fake-link"
  data-testid="assign-reviewers"
  id="assign-reviewers"
  onClick={[Function]}
  type="button"
>
  Assign
  <span
    className="govuk-visually-hidden"
  >
     
    reviewers
  </span>
</button>
`;

exports[`WorkbasketUserAssignment renders assign workers button 1`] = `
<button
  className="govuk-link fake-link"
  data-testid="assign-workers"
  id="assign-workers"
  onClick={[Function]}
  type="button"
>
  Assign
  <span
    className="govuk-visually-hidden"
  >
     
    workers
  </span>
</button>
`;

exports[`WorkbasketUserAssignment renders unassign reviewers button 1`] = `
<button
  className="govuk-link fake-link"
  data-testid="unassign-reviewers"
  id="unassign-reviewers"
  onClick={[Function]}
  type="button"
>
  Unassign
  <span
    className="govuk-visually-hidden"
  >
     
    reviewers
  </span>
</button>
`;

exports[`WorkbasketUserAssignment renders unassign workers button 1`] = `
<button
  className="govuk-link fake-link"
  data-testid="assign-workers"
  id="assign-workers"
  onClick={[Function]}
  type="button"
>
  Unassign
  <span
    className="govuk-visually-hidden"
  >
     
    workers
  </span>
</button>
`;
